### PR TITLE
[Cosmos DB] az cosmosdb create/update: Improve error message from incorrect --locations input

### PR DESCRIPTION
**Pull request by shurd**
**Description**  
Updates the error message user sees when supplying invalid key name for the "locations" parameter during update or create. Helps address issue https://github.com/Azure/azure-cli/issues/14627.

Fix bug preventing removing ip range filter via command line (handle empty string). Helps address issue https://github.com/Azure/azure-cli/issues/15291.

Add analytical storage ttl support for SQL containers.

Fixed typo in help text to address https://github.com/Azure/azure-cli/issues/15524

**Testing Guide**  
az cosmosdb update -g {g} -n {n} --locations regionName=EastUS failoverPriorit=0
az cosmosdb update -g {g} -n {n} --ip-range-filter ""

**History Notes**  

[Cosmos DB] az cosmosdb create/update: Improve error message from incorrect --locations input
[Cosmos DB] az cosmosdb sql container create/update: Add --analytical-storage-ttl parameter

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Cosmos

**Comment by shurd** [comment]
@Juliehzl can you please take a look?

**Comment by yonzhan** [comment]
@Juliehzl will review this PR after Chinese National holiday.

File: src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py [hunk 1]
```diff
@@ -154,6 +154,7 @@ def load_arguments(self, _):
         c.argument('conflict_resolution_policy', options_list=['--conflict-resolution-policy', '-c'], type=shell_safe_json_parse, completer=FilesCompleter(), help='Conflict Resolution Policy, you can enter it as a string or as a file, e.g., --conflict-resolution-policy @policy-file.json or ' + SQL_GREMLIN_CONFLICT_RESOLUTION_POLICY_EXAMPLE)
         c.argument('max_throughput', max_throughput_type)
         c.argument('throughput', help='The throughput of SQL container (RU/s). Default value is 400. Omit this parameter if the database has shared throughput unless the container should have dedicated throughput.')
+        c.argument('analytical_storage_ttl', type=int, help='Analytical TTL, when analytical storage is enabled.')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py:157]: Is there a default value in server side?

**Comment by Juliehzl** [comment]
@shurd could you take a look for CI failure?

File: src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py (hunk 1 above)
Comment (shurd) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py:157]: The default value is None/null I believe

**Comment by shurd** [comment]
@Juliehzl The CI has two failures. One is saying the parameter "analytical_storage_ttl" is too long for the commands I just added it on; however, this already exists for several other commands (mongodb collection create/update). Is there a way to suppress this error? Why is CI not marking the other places as problematic.

The other CI error is saying that "cosmosdb collection/database" commands have resource-group-name as a parameter, which is not allowed. These commands are all deprecated, so can we suppress this error?

**Comment by Juliehzl** [comment]
> @Juliehzl Zunli Hu FTE The CI has two failures. One is saying the parameter "analytical_storage_ttl" is too long for the commands I just added it on; however, this already exists for several other commands (mongodb collection create/update). Is there a way to suppress this error? Why is CI not marking the other places as problematic.
> 
> The other CI error is saying that "cosmosdb collection/database" commands have resource-group-name as a parameter, which is not allowed. These commands are all deprecated, so can we suppress this error?

For the first error, you could add options_list=['--analytical-storage-ttl', '-t'] to bypass it, which means you need to add one more short option for the parameter.

**Comment by Juliehzl** [comment]
The second one is not blocking issue. You could not change  anything for it. But it would be better if we could have an ETA to remove these deprecated commands.

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**